src/metabase/models/field.clj

```diff
@@ -209,6 +210,18 @@
     (u/key-by :field_id (when (seq field-ids)
                           (db/select model :field_id [:in field-ids])))))
 
+(defn nfc-field->parent-identifier
+  "Take a nested field column field corresponding to something like an inner key within a JSON column,
+  and then get the parent column's identifier from its own identifier and the nfc path stored in the field"
```

Comment:
this took me a minute to understand...up to you, but maybe an illustrative example would be helpful for folks who aren't super familiar with this code